tensorflow/compiler/xla/service/gpu/cudnn_conv_runner.cc

```diff
@@ -80,7 +80,16 @@ Status RunCudnnConvImpl(const CudnnConvParams& params,
   auto output_buf = se::DeviceMemory<T>(params.output_buf);
   AlgorithmConfig algorithm = params.algorithm;
 
-  if (options.algo_override) {
+  // in ROCm mode, the first call to run the convolution needs to trigger the
+  // code that calls miopenFind* API. That triggger is implicit, it is based
+  // on whether or not the AlgorithmConfig::algorithm is empty! So for the
+  // first call we need to ensure that the AlgorithmConfig::algorithm is
+  // empty. For all subsequent calls, we should use the value retrieved from
+  // the backend_config
+  if ((options.algo_override.has_value()) &&
+      (*options.algo_override == se::dnn::AlgorithmDesc())) {
```

Comment:
@timshen91 FYI, I did this instead. When I take a second look at the interface, it does not make sense to expose the algorithm as a separate option for `ROCm` because the options argument already have one. So I create a empty algorithm as a signal to `ROCm` code base that this is the first time it is called.

Reply:
Also as an FYI, the best way to review the code changes now is to look directly at the files changed across all commits